src/diffusers/modeling_utils.py

```diff
@@ -82,12 +87,16 @@ def find_tensor_attributes(module: torch.nn.Module) -> List[Tuple[str, Tensor]]:
         return first_tuple[1].dtype
 
 
-def load_state_dict(checkpoint_file: Union[str, os.PathLike]):
+def load_state_dict(checkpoint_file: Union[str, os.PathLike], is_pytorch: bool):
     """
     Reads a PyTorch checkpoint file, returning properly formatted errors if they arise.
     """
     try:
-        return torch.load(checkpoint_file, map_location="cpu")
+        if is_pytorch:
```

Comment:
```suggestion
        if Path(checkpoint_file).name == WEIGHTS_NAME:
```

Reply:
I feel this is wrong, we already know at this point what kind of file it is so we don't have to guess.

I agree the `is_pytorch` is slightly odd but I don't think doing things based on filename is great. 
I'll try to figure something out that's better.

Reply:
Couldn't find anything satisfactory to me so I went your way.